### PR TITLE
feat(@turbo/repository): return dependencies in graph

### PR DESCRIPTION
**Pull request by mehulkar**
The test changes are just to break up the files so jest can parallelize execution.

Closes TURBO-2524

### CODE REVIEW

**Review by gaspar09 — APPROVED** [review]

File: packages/turbo-repository/js/index.d.ts [hunk 1]
```diff
@@ -32,7 +36,7 @@ export class Workspace {
    * Finds and returns a map of packages within the workspace and its
    * dependents (i.e. the packages that depend on each of those packages).
    */
-  findPackagesAndDependents(): Promise<Record<string, Array<string>>>;
+  findPackagesWithGraph(): Promise<SerializablePackages>;
```
Comment (gaspar09) [inline packages/turbo-repository/js/index.d.ts:39]: Need to update the comment

File: packages/turbo-repository/rust/src/lib.rs [hunk 2]
```diff
@@ -110,26 +144,41 @@ impl Workspace {
...
+                return Err(Error::from_reason(e.to_string()));
+            }
         };
 
-        let map: HashMap<String, Vec<String>> = packages
+        // Create a map of package names to their dependents and dependencies
+        //  {
+        //    "package-path": {
+        //      "dependents": ["dependent1_path", "dependent2_path"],
+        //      "dependencies": ["dependency1_path", "dependency2_path"]
```
Comment (gaspar09) [inline packages/turbo-repository/rust/src/lib.rs:162]: Does this include non-workspace dependencies?

**Review by gaspar09 — CHANGES_REQUESTED** [review]

File: packages/turbo-repository/rust/src/lib.rs (hunk 2 above)
Comment (mehulkar) [reply packages/turbo-repository/rust/src/lib.rs:162]: What do you mean? These will be the list of _packages_ in the workspace, but just their relative paths relative to the root of the workspace

File: packages/turbo-repository/js/index.d.ts (hunk 1 above)
Comment (mehulkar) [reply packages/turbo-repository/js/index.d.ts:45]: done

File: packages/turbo-repository/rust/src/lib.rs (hunk 2 above)
Comment (gaspar09) [reply packages/turbo-repository/rust/src/lib.rs]: Yea that's what I meant.

Comment (gaspar09) [reply packages/turbo-repository/rust/src/lib.rs]: The api kinda misleads to make it seem like you return all dependencies not just the ones in the same workspace. 
But I guess it's just something you read in the docs once.

**Review by gaspar09 — APPROVED** [review]

File: packages/turbo-repository/rust/src/lib.rs [hunk 3]
```diff
@@ -107,29 +141,43 @@ impl Workspace {
...
-        &self,
-    ) -> Result<HashMap<String, Vec<String>>, Error> {
+    pub async fn find_packages_with_graph(&self) -> Result<SerializablePackages, Error> {
         let packages = self.find_packages().await?;
 
         let workspace_path = match AbsoluteSystemPath::new(self.absolute_path.as_str()) {
             Ok(path) => path,
-            Err(e) => return Err(Error::from_reason(e.to_string())),
+            Err(e) => {
+                print!("Error: {:?}", e);
```
Comment (chris-olszewski) [inline packages/turbo-repository/rust/src/lib.rs:159]:
```suggestion
```

**Review by chris-olszewski — APPROVED** [review]

File: packages/turbo-repository/rust/src/lib.rs [hunk 4]
```diff
@@ -73,13 +84,36 @@ impl Package {
         workspace_path: &AbsoluteSystemPath,
     ) -> Vec<Package> {
         let node = PackageNode::Workspace(PackageName::Other(self.name.clone()));
-        let ancestors = match graph.immediate_ancestors(&node) {
-            Some(ancestors) => ancestors,
+        let pkgs = match graph.immediate_ancestors(&node) {
```
Comment (gsoltis) [inline packages/turbo-repository/rust/src/lib.rs:87]: Are you sure you want just `immediate_ancestors`? What about transitive dependents? If you do in fact just want immediates, maybe a comment or update the method name to indicate that.

File: packages/turbo-repository/rust/src/lib.rs [hunk 5]
```diff
@@ -73,13 +84,36 @@ impl Package {
...
+            .collect()
+    }
+
+    fn dependencies(
+        &self,
+        graph: &PackageGraph,
+        workspace_path: &AbsoluteSystemPath,
+    ) -> Vec<Package> {
+        let node = PackageNode::Workspace(PackageName::Other(self.name.clone()));
+        let pkgs = match graph.immediate_dependencies(&node) {
```
Comment (gsoltis) [inline packages/turbo-repository/rust/src/lib.rs:110]: As above, this is getting just immediate dependencies, not the full transitive closure. Might be worth a comment at least.

File: packages/turbo-repository/rust/src/lib.rs (hunk 3 above)
Comment (gsoltis) [inline packages/turbo-repository/rust/src/lib.rs:159]: Probably should remove this line.

**Review by gsoltis — APPROVED** [review]
One errant `print!` and a couple clarifications, but otherwise LGTM

File: packages/turbo-repository/rust/src/lib.rs (hunk 4 above)
Comment (mehulkar) [reply packages/turbo-repository/rust/src/lib.rs:87]: just immediate for now. We are constructing the graph client side in our use case, we'll keep doing that for now and iterate here. I can't remember why I didn't do full tree on `ancestors` when I first did it, probably Rust-related hurdles.

**Pull request merged**